boards/arm/sama5/giant-board/src/sam_ethernet.c

```diff
@@ -0,0 +1,328 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+/* Force verbose debug on in this file only to support unit-level testing. */
+
+#ifdef CONFIG_NETDEV_PHY_DEBUG
+#  undef  CONFIG_DEBUG_INFO
+#  define CONFIG_DEBUG_INFO 1
+#  undef  CONFIG_DEBUG_NET
+#  define CONFIG_DEBUG_NET 1
+#endif
+
+#include <string.h>
+#include <assert.h>
+#include <debug.h>
+
+#include <nuttx/irq.h>
+#include <nuttx/arch.h>
+
+#include "sam_pio.h"
+#include "sam_ethernet.h"
+
+#include "giant-board.h"
+
+#ifdef HAVE_NETWORK
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifndef CONFIG_SAMA5_EMACA
+#  undef CONFIG_SAMA5_EMAC_ISETH0
+#endif
+
+#ifdef CONFIG_SAMA5_EMAC_ISETH0
+#  define SAMA5_EMAC_DEVNAME "eth0"
+#  define SAMA5_GMAC_DEVNAME "eth1"
+#else
+#  define SAMA5_GMAC_DEVNAME "eth0"
+#  define SAMA5_EMAC_DEVNAME "eth1"
+#endif
+
+/* Debug ********************************************************************/
+
+/* Extra, in-depth debug output that is only available if
+ * CONFIG_NETDEV_PHY_DEBUG us defined.
+ */
+
+#ifdef CONFIG_NETDEV_PHY_DEBUG
+#  define phyerr    _err
+#  define phywarn   _warn
+#  define phyinfo   _info
+#else
+#  define phyerr(x...)
+#  define phywarn(x...)
+#  define phyinfo(x...)
+#endif
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: sam_emac_phy_enable and sam_gmac_enable
+ ****************************************************************************/
+
+#ifdef CONFIG_SAMA5_PIOE_IRQ
+#ifdef CONFIG_SAMA5_EMACA
+static void sam_emac_phy_enable(bool enable)
+{
+  phyinfo("IRQ%d: enable=%d\n", IRQ_INT_ETH1, enable);
+  if (enable)
+    {
+      sam_pioirqenable(IRQ_INT_ETH1);
+    }
+  else
+    {
+      sam_pioirqdisable(IRQ_INT_ETH1);
+    }
+}
+
+#endif
+
+#ifdef CONFIG_SAMA5_GMAC
+static void sam_gmac_phy_enable(bool enable)
+{
+  phyinfo("IRQ%d: enable=%d\n", IRQ_INT_ETH0, enable);
+  if (enable)
+    {
+      sam_pioirqenable(IRQ_INT_ETH0);
+    }
+  else
+    {
+      sam_pioirqdisable(IRQ_INT_ETH0);
+    }
+}
```

Comment:
The SAMD2 does not have a gigabit MAC, does it?  I think this needs to be cleaned up.  It has 2x100mbit MACs, right?

The naming is confusing because some Atmel documentation refers to the 100mbit MAC as a GMAC.  The code refers to these and EMAC.  I think you need to remove all references to GMAC unless I am not understanding.

Reply:
@patacongo SAMA5D2 has a GMAC peripheral that is a single 10 or 100 Mbps Ethernet. I will look at the Atmel example code to see whether it is a GMAC or EMAC, then update the references.

Reply:
@patacongo Since the board does not bring out the Ethernet pins, will remove the file.